server/src/main/java/org/elasticsearch/index/engine/ReadOnlyEngine.java

```diff
@@ -575,7 +576,9 @@ public void advanceMaxSeqNoOfUpdatesOrDeletes(long maxSeqNoOfUpdatesOnPrimary) {
 
     protected DirectoryReader openDirectory(Directory directory) throws IOException {
         assert Transports.assertNotTransportThread("opening directory reader of a read-only engine");
-        final DirectoryReader reader = DirectoryReader.open(directory);
+        var minVersion = engineConfig.getIndexSettings().getIndexMetadata().getCompatibilityVersion().luceneVersion().major;
+        var commit = Lucene.getIndexCommit(SegmentInfos.readLatestCommit(directory, minVersion), directory);
+        final DirectoryReader reader = DirectoryReader.open(commit, minVersion, null);
```

Comment:
here too, I would hardcode the version lower bound to be based on the lucene major of the minimum read only compatible index version.

Reply:
Sure.